Update Kotlin compiler to escape package names

pkwarren opened the pull request:
The current Kotlin code generator creates code which fails to compile with imports like:

```
syntax = "proto3";
package net.example.v1;
message Sample {
  string net = 1;
}
```

This results in a condition where 'net' is ambiguous in some cases (could refer to the field or the package). To avoid these issues, the code generator should enclose package paths in backticks:

```
net.example.v1.SampleKt => `net.example.v1`.SampleKt
```

Fixes #13182.

acozzette commented:
@pkwarren Could you merge the current `main` into your branch (or rebase if you prefer)? I think the test failures are things that were fixed recently on main but those fixes might not take effect until you update your branch.

pkwarren commented:
> @pkwarren Could you merge the current `main` into your branch (or rebase if you prefer)? I think the test failures are things that were fixed recently on main but those fixes might not take effect until you update your branch.

Updated.

jhump commented:
@zhangsk, @fowles, anything else needed before this PR can be merged?

zhangskz commented:
Thanks for checking in -- looks like this got stuck when going through the internal submit process due to some failing tests that seem to be unrelated.

Giving it a kick but please feel free to ping again if you don't see this submitted soon. Sorry for the delay!

pkwarren commented:
@zhangskz, @fowles, @acozzette - Anything remaining I need to do on my end to get this PR merged? Happy to merge latest main/rebase+squash if needed.

zhangskz commented:
Looks like the change for this somehow got garbage collected internally after getting stuck on already failing tests -- we gave it another kick and hopefully it should go through properly this time.

jhump commented:
@zhangskz, any updates? Can this be merged yet?

pkwarren commented:
I've gone ahead and rebased against latest main and squashed down to a single commit (in case that was preventing this from being automatically merged).
